internal/pkg/cli/task_run.go

```diff
@@ -134,6 +163,117 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute create or update task resources and run the task.
+func (o *runTaskOpts) Execute() error {
+	if o.image == "" && o.dockerfilePath == "" {
+		o.dockerfilePath = "./Dockerfile"
+	}
+
+	if o.imageTag == "" {
+		o.imageTag = "latest"
+	}
+
+	if err := o.getNetworkConfig(); err != nil {
+		return err
+	}
+
+	if err := o.deployTaskResource(); err != nil {
```

Comment:
Got confused by this: why do we deploy task resource here? and deploy again https://github.com/aws/copilot-cli/pull/1077/files#diff-d83adc990a0da087a39c02967f267ed3R193

Reply:
'deploy' may be a bad name for this. The first time it's called is to create all the resources, and the second time is to update the resources. This is because we need to create a ECR repo first, and then if we are not using an existing image that the user provides, we push the image to the repo, and then update that pushed image to the task definition.

Reply:
oooh got it. Yeah agree on renaming the method name. It is deploying and then updating under the hood.